### ISSUE TITLE
Allow backup of input file

### ISSUE THREAD

**Issue by ghost** [chmln/sd#4]
`-b` flag, which creates a `inputfile.bak` backup before replacing.

**Comment by rahilwazir** [comment]
It will be great. But I think `-o` output flag will be more beneficial to create new file along with a choice to rename it as well. i.e.

```
❯ cat url.txt | sd -s 'google.com' 'google.test' -o url2.txt
```

Should output `url2.txt` with replaced string. But at the same time it will easily be achieved via stdout 

```
❯ cat url.txt | sd -s 'google.com' 'google.test' > url2.txt
```

**Comment by chmln** [comment]
I believe backing up files before replacing them is an edge case for the majority of people. If someone is worried about making the wrong replacements, is easy enough to just use pipes and create a new file instead. e.g. `sd abc def < important_file.txt > important_file.new.txt`

**Comment by AndydeCleyre** [comment]
I disagree about it being "easy enough" to use pipes instead, especially because

1. your example doesn't even accomplish the desired behavior here; the new content should be in the old filename, and the original content should be in the new filename. The way you have it requires extra work afterward to get to the final state, whereas the desired behavior only requires paying any attention to the backup file if something goes wrong/surprising.
2. if you want to do this on a huge tree of globbed files, you'll have to make a loop that carefully manages each placement, and this increases the probability of user error by a lot

Thanks for the great project, btw!

**Comment by nc7s** [comment]
More or less a rant, but:

```
for file in *; do
  cp "$file" "$file.old"
  sd FIND REPLACE "$file"
done
```

Also:

```
cp file.txt file.txt.old && sd FIND REPLACE file.txt
```

Keystrokes for the 3 occurrences of `file.txt` can be reduced with a decent tab completion or a prefixing env var.
